woocommerce/woocommerce
# Add documentation for cache purging with "coming soon" template edits

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/50077.

This PR updates the documentation to help hosting providers integrate WooCommerce’s Coming Soon mode more effectively. The new section details how to use the `save_post_wp_template`, `save_post_wp_template_part` and `save_post_wp_global_styles` (https://developer.wordpress.org/reference/hooks/save_post_post-post_type/) hooks to detect edits to the "coming soon" template and trigger cache purges. By implementing these hooks, hosts can ensure template updates take effect immediately without requiring additional manual steps.

Additionally, this PR updates the documentation that we're now using 60s cache for coming soon pages.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Review the documentation

1. Install `Code Snippets` plugin
2. Add the following snippet

```php
add_action( 'save_post_wp_template', 'purge_cache_on_template_change', 10, 3 );
add_action( 'save_post_wp_template_part', 'purge_cache_on_template_change', 10, 3 );
add_action( 'save_post_wp_global_styles', 'purge_cache_on_template_change', 10, 3 );

function purge_cache_on_template_change( $post_id, $post, $update ) {
    // Check if the template is associated with the coming soon mode.
    if ( 'coming-soon' === $post->post_name ) {
       error_log( 'coming soon template changed' );
    }
}
```

3. Go to `/wp-admin/site-editor.php?postType=wp_template&postId=woocommerce/woocommerce//coming-soon&canvas=edit`

4. Make some changes to the template (e.g. add a new block, change global styles)
5. Check the debug.log and see that `coming soon template changed` is logged



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
